api/keyconjurer/serverless_functions.go

```diff
@@ -87,13 +89,13 @@ func (h *Handler) GetUserDataEventHandler(ctx context.Context, req *events.ALBTa
 	}
 
 	log = h.log.WithFields(logrus.Fields{"username": event.Credentials.Username, "idp": event.AuthenticationProvider})
-	provider, ok := h.authenticationProviders.Get(event.AuthenticationProvider)
+	provider, ok := providers.Get(event.AuthenticationProvider)
 	if !ok {
 		log.Infof("unknown provider %q", provider)
 		return ErrorResponse(ErrCodeInvalidProvider, "the provider you supplied is unsupported by this version of KeyConjurer")
 	}
 
-	user, err := provider.Authenticate(ctx, event.Credentials)
+	user, err := provider.Authenticate(ctx, providers.Credentials(event.Credentials))
```

Comment:
This separate type is used to distinguish between `core.Credentials`, which:

* Can be encrypted
* Are the representation the public-facing API uses

From `providers.Credentials`, which might not always be encrypted.

Reply:
maybe we need a more distinguishable name between the two?

Reply:
Originally I did have `EncryptedCredentials` but they aren't always encrypted (For example, in the `login` call).

I don't think we need to yakshave too much about this. A line of documentation indicating the difference will be sufficient - `core.Credentials` and `provider.Credentials` will be going away in 3 months with Yubi support.